pkg/collector/corechecks/oracle-dbm/sql_wrappers.go

```diff
@@ -8,17 +8,57 @@
 package oracle
 
 import (
+	"fmt"
+	"strings"
+
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 	"github.com/jmoiron/sqlx"
-	"strings"
 )
 
 func selectWrapper[T any](c *Check, s T, sql string, binds ...interface{}) error {
 	err := c.db.Select(s, sql, binds...)
-	reconnectOnConnectionError(c, &c.db, err)
+	err = handleError(c, &c.db, err)
+	return err
+}
+
+func getWrapper[T any](c *Check, s T, sql string, binds ...interface{}) error {
+	err := c.db.Get(s, sql, binds...)
+	err = handleError(c, &c.db, err)
+	return err
+}
+
+func handleError(c *Check, db **sqlx.DB, err error) error {
+	if err == nil {
+		return err
+	}
+	err, isPrivilegeError := handlePrivilegeError(c, err)
+	if err != nil && isPrivilegeError {
+		return err
+	}
+	reconnectOnConnectionError(c, db, err)
 	return err
 }
 
+func handlePrivilegeError(c *Check, err error) (error, bool) {
+	var isPrivilegeError bool
+	if err == nil {
+		return nil, isPrivilegeError
+	}
+	if !strings.Contains(err.Error(), "ORA-00942") {
```

Comment:
Generally, we can't, because `ORA-00942` could really mean that we misspelled the view name. But in the production, we can eliminate this possibility, so the only conceivable cause is really a missing privilege.